pylanetary/spectral/core.py

```diff
@@ -0,0 +1,808 @@
+  test_cube.wind_calc(picklefile = 'development_testing_fit.pickle', restfreq = 345.79598990, outfile = 'development_testing_wind')
+  test_cube.plot_data(picklefile = 'development_testing_wind.pickle', platescale = 0.1, body = 'Neptune', title = 'CO Doppler Velocity Map', cont_label = 'Radial Velocity (m/s)', date = '2016-04-30 00:00', location = 'ALMA', spatial = 4e4, limits = [-2000,2000], contours = [21,200])
+  
+  '''
+  ###############################################################################
+  
+  def __init__(self, imagepath, x_space):
+    '''
+    Input string instantiates object using the FITS image path.
+    
+    Parameters
+    ----------
+    imagepath : string
+          The path to FITS image file.
+    x_space : string
+          The identifier for choosing wavelength or frequency space for the model fit. Input 'F' or 'f' for frequency space, 'W' or 'w' for wavelength space.
+    
+    Attributes
+    ----------
+    chan_num : float
+          The number of frequency channels. Pulled from image header.
+    i0 : float
+          The reference pixel of the image. Pulled from image header.
+    df : float
+          The channel width in Hz. Pulled from image header.
+    f0 : float
+          The rest frequency of the image in Hz. Pulled from image header.
+    xpixmax : float
+          The x-dimension size in pixels. Pulled from image header.
+    ypixmax : float
+          The y-dimension size in pixels. Pulled from image header.
+    bmaj : float
+          The semi-major axis of the restoring beam in arcsec. Pulled from image header. 
+    bmin : float
+          The semi-minor axis of the restoring beam in arcsec. Pulled from the image header.
+    bpa : float
+          The position angle of the restoring beam in degrees. Pulled from the image header.
+    data : numpy array
+          The spectral data pulled from the image cube.
+    freqspec : numpy array
+          The frequency grid around the rest frequency, which can be used as the x-axis.
+    wl_spec : numpy array
+          The wavelength grid around the rest wavelength, which can be used as the x-axis.
+    
+    '''  
+    self.x_space = x_space
+    
+    fits_file = fits.open(imagepath)
+    hdr = fits_file[0].header
+    
+    self.chan_num = hdr['NAXIS3'] 
+    self.i0 = hdr['CRPIX3'] 
+    self.df = hdr['CDELT3'] 
+    self.f0 = hdr['CRVAL3']
+    self.xpixmax = hdr['NAXIS1'] 
+    self.ypixmax = hdr['NAXIS2']
+    self.bmaj = hdr['BMAJ']
+    self.bmin = hdr['BMIN']
+    self.bpa = hdr['BPA']
+    self.data = fits_file[0].data
+    fits_file.close()
+    
+    print('Number of pixels: '+str(self.xpixmax)+','+str(self.ypixmax))
+    # Correct for FITS 1-indexing scheme
+    if self.i0 < 0: 
+      self.i0 = self.i0 - 1
+    else:
+      self.i0 = self.i0 + 1
+    
+    self.freqspec = ((np.arange(self.chan_num) - self.i0) * self.df + self.f0)/1e9
+    self.wl_spec = c/self.freqspec
+      
+    if self.x_space == 'F' or self.x_space == 'f':
```

Comment:
same as before, I don't think we should be opinionated about this just yet in the __init__.  My vision for the spectral subpackage is that eventually these data cubes could be used for a million different things, not just lmfit, so we want to only set attributes that are generic to the idea of a spectral data cube.